inspirehep/utils/citations.py

```diff
@@ -22,39 +22,44 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-from invenio_ext.template import render_template_to_string
+from flask import jsonify
 
+from invenio_ext.template import render_template_to_string
 
-class Citation(object):
-    """Class used to output citations format in detailed record"""
+from invenio_search.api import Query
 
-    def __init__(self, record):
-        self.record = record
 
-    def citations(self):
-        """Citation export for single record in datatables format.
+def render_citations(recid):
+    """Citation export for single record in datatables format.
 
         :returns: list
             List of lists where every item represents a datatables row.
             A row consists of [reference, num_citations]
-        """
-        from invenio_search.api import Query
+    """
+    out = []
+    row = []
+    es_query = Query('refersto:' + str(recid)).search()
+    es_query.body.update({
+        'sort': [{'citation_count': {'order': 'desc'}}],
+        'size': 9999
+    })
+    citations = es_query.records()
 
-        out = []
+    for citation in citations:
+        row.append(render_template_to_string("citations.html",
+                                             record=citation, reference=None))
+        row.append(citation.get('citation_count', ''))
+        out.append(row)
         row = []
-        recid = self.record['control_number']
-        es_query = Query('refersto:' + str(recid)).search()
-        es_query.body.update({
-            'sort': [{'citation_count': {'order': 'desc'}}],
-            'size': 9999
-        })
-        citations = es_query.records()
-
-        for citation in citations:
-            row.append(render_template_to_string("citations.html",
-                                                 record=citation))
-            row.append(citation.get('citation_count', ''))
-            out.append(row)
-            row = []
-
-        return out
+
+    return out
+
+
+def get_ajax_citations(recid):
+    """Jsonified citation export for single record in datatables format."""
+
+    return jsonify(
```

Comment:
Perhaps this should ideally be placed in the `views: ajax_citations` instead, so that it is clear in the views what is returned. That would make this function go away and instead the views will be:

```python
from inspirehep.utils.citations import render_citations
return jsonify({
    "data": render_citations(recid)
})
```
Same with `ajax_references`.